src/tox/session/env_select.py

```diff
@@ -241,13 +242,29 @@ def _finalize_config(self) -> None:
         self._state.conf.core.mark_finalized()
 
     def _build_run_env(self, name: str) -> RunToxEnv | None:
-        if self._provision is not None and self._provision[0] is False and name == self._provision[1]:
+        provision_on = provision_tox_env = provision_loader = None
+        if self._provision is not None:
+            provision_on, provision_tox_env, provision_loader = self._provision
+        if not provision_on and name == provision_tox_env:
+            # not a provision run, so don't include provision env (.tox) as a run env
+            return None
+        if provision_on and name != provision_tox_env:
+            # for provision run, don't consider any environment other than provision env
             return None
         env_conf = self._state.conf.get_env(name, package=False)
-        if self._provision is not None and self._provision[1] == name:
-            env_conf.loaders.insert(0, self._provision[2])
-        desc = "the tox execute used to evaluate this environment"
-        env_conf.add_config(keys="runner", desc=desc, of_type=str, default=self._state.conf.options.default_runner)
+        runner_key_desc = "the tox execute used to evaluate this environment"
+        if name == provision_tox_env:
+            if provision_loader is not None:
+                env_conf.loaders.insert(0, provision_loader)
+            # provision environment always runs with VirtualEnvRunner
+            env_conf.add_constant(keys="runner", desc=runner_key_desc, value=VirtualEnvRunner.id())
```

Comment:
Virtualenv should be the default but users should be able to override it within .tox testenv section.